xbmc/filesystem/posix/PosixFile.cpp

```diff
@@ -313,6 +314,18 @@ bool CPosixFile::Rename(const CURL& url, const CURL& urlnew)
   
   if (rename(name.c_str(), newName.c_str()) == 0)
     return true;
+
+  // rename across mount points - need to copy/delete
+  if (errno == EXDEV)
+  {
+    if (XFILE::CFile::Copy(name, newName))
+    {
+      if (XFILE::CFile::Delete(name))
```

Comment:
why? if delete failed, move/rename failed. It'd be wrong to return true and leave copies.